modules/store/src/selector.ts

```diff
@@ -122,34 +122,19 @@ export function defaultMemoize(
   return { memoized, reset, setResult, clearResult };
 }
 
-export function createSelector<State, S1, Result>(
-  s1: Selector<State, S1>,
-  projector: (s1: S1) => Result
+export function createSelector<State, S extends unknown[], Result>(
+  ...args: [...Selector<State, unknown>[], unknown] &
+    [...{ [i in keyof S]: Selector<State, S[i]> }, (...s: S) => Result]
 ): MemoizedSelector<State, Result>;
```

Comment:
one might think that this would be sufficient:

```typescript
export function createSelector<State, S extends unknown[], Result>(
  ...args: [...{ [i in keyof S]: Selector<State, S[i]> }, (...s: S) => Result]
): MemoizedSelector<State, Result>;
```

however, TS refuses to match on this override with that as it always infers `State` as `unknown`.

the best I came up with to resolve that was to try to infer `State` by including a type definition specifically for it.